wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -569,6 +575,26 @@ private WorkspaceImpl doCreateWorkspace(
     return workspace;
   }
 
+  /**
+   * If 'generateName' is defined and 'name' is not, we generate name using 'generateName' as a
```

Comment:
Could we use `generateName` all the time if it is defined ?

Reply:
kubernetes takes `name` in that case and we follow kubernetes approach here.